crates/transaction-pool/src/pool/best.rs

```diff
@@ -61,6 +62,14 @@ pub(crate) struct BestTransactions<T: TransactionOrdering> {
     pub(crate) independent: BTreeSet<PendingTransaction<T>>,
     /// There might be the case where a yielded transactions is invalid, this will track it.
     pub(crate) invalid: HashSet<TxHash>,
+    /// Used to recieve any new pending transactions that have been added to the pool after this
+    /// iterator was snapshotted
+    ///
+    /// We prioritize these new transactions over existing ones by first inserting these into the
+    /// pool, and then yielding. This breaks the contract that the iterator yields the directly
+    /// "next" best transaction from the POV of the consumer, but we assume more often than not
+    /// newer transactions are better
```

Comment:
let's move these docs to the trait docs, that there is no guarantee that the iterator will only ever return txs with decreasing priority, and instead always returns the best transaction that it currently knows